src/api/commands/priceticker/PriceTickerRootCommand.ts

```diff
@@ -9,24 +9,53 @@ import { BaseCommand } from '../BaseCommand';
 import { RpcCommandFactory } from '../../factories/RpcCommandFactory';
 import { Commands } from '../CommandEnumType';
 
+import { PriceTickerService } from '../../services/PriceTickerService';
+import { MessageException } from '../../exceptions/MessageException';
+import { PriceTicker } from '../../models/PriceTicker';
+
+
 export class PriceTickerRootCommand extends BaseCommand implements RpcCommandInterface<void> {
 
     public log: LoggerType;
 
     constructor(
-        @inject(Types.Core) @named(Core.Logger) public Logger: typeof LoggerType
+        @inject(Types.Core) @named(Core.Logger) public Logger: typeof LoggerType,
+        @inject(Types.Service) @named(Targets.Service.PriceTickerService) private priceTickerService: PriceTickerService
     ) {
         super(Commands.PRICETICKER_ROOT);
         this.log = new Logger(__filename);
     }
 
     @validate()
     public async execute( @request(RpcRequest) data: RpcRequest, rpcCommandFactory: RpcCommandFactory): Promise<any> {
-        return await this.executeNext(data, rpcCommandFactory);
+        if (data.params.length > 0) {
+            const currencies = data.params;
+            let returnData: any = [];
+            for (const currency of currencies) { // INR, USD
+                let searchData: any = await this.priceTickerService.search(currency);
```

Comment:
move the implementation to the service layer.
this.priceTickerService.getPriceTickers(currencies);